py/scheduler.c

```diff
@@ -30,6 +30,17 @@
 
 #if MICROPY_ENABLE_SCHEDULER
 
+static inline bool mp_sched_full(void) {
+    // MICROPY_SCHEDULER_DEPTH must be a power of 2
```

Comment:
This is a new restriction, although probably a fair one to impose (existing uses in tho repo are a power of 2).

Reply:
That was a hard requirement of the previous head/tail rolling approach, and of the masking used in this newer approach... but not really necessary in an idx/len algorithm so just as easy to remove it again I'd say